docs/extending-nsm.md

```diff
@@ -0,0 +1,71 @@
+# Extending Network Service Mesh (NSM)
+
+
+# Introduction
+
+This document is initially aimed at developers looking to extend the NSM with new plugins to provide users with new functionality. It is based on the version of NSM available August 2018 in the [Network Service Mesh Github](https://github.com/ligato/networkservicemesh) Repository. It documents the steps that were used to create the simple-dataplane extension. 
+
+All steps assume the top level directory is networkservicemesh from github.
+
+
+# Development Steps
+
+The development steps are to add a new data plane called cnf-dataplane.
+
+
+## Edit Makefile
+
+The top level Makefile [networkservicemesh/Makefile] to add to the following targets
+
+docker-build: add "docker-build-cnf-dataplane"
+
+docker-push: add "docker-push-cnf-dataplane"
+
+
+## Dockerfile
+
+A new dockerfile is added: networkservicemesh/build/Dockerfile-cnf.dataplane
+
+
+## API Directory and Interface Definition
+
+Create a new directory for the cnf-dataplane daemonset CRD interface definition.
+
+The new directory is: networkservicemesh/pkg/nsm/apis/cnfdataplane
+
+The interface definition file written in Google Protobufs is added to this directory. 
+
+The file will be:  networkservicemesh/pkg/nsm/apis/cnfdataplane/cnfdataplane.proto
+
+The interface definitions must match the go code created for the daemonset.
```

Comment:
I am not sure if it is correct to give an accent on daemonset, potentially anything even a pod could be a plugin. I would suggest to use a more generic term here.